### PR TITLE
Ignore AcquireLocalTicket errors

### LINKED ISSUE #952: "ServerFaultCode: The operation is not supported on the object"
I've used govc successfully in the past, and I have production jobs using it regularly, but today I'm trying to use govc with my homelab cluster, which it doesn't seem to like:

```
bard:~ $ govc ls
govc: ServerFaultCode: The operation is not supported on the object.
bard:~ $ govc about
govc: ServerFaultCode: The operation is not supported on the object.
```

Does that mean my VMUG EVALExperience license for my 3-node vSphere + vCenter HA setup doesn't include API access?

If so, can we get a better error message in govc telling users what the actual problem is?

Or do I need to enable the API server in vCenter somewhere? I've only ever worked with vSphere clusters others have created before. This home one is the first I've put together, and perhaps I screwed it up.

Sorry for the noob questions.

Thanks!

### PR DESCRIPTION
**Pull request by dougm**
If no username is provided, the govc client tries to acquire a local ticket.
When invoked remotely, ESX returns an InvalidRequestFault (VC returns NotSupportedFault).
So, rather than return an error here, fallthrough to Login() with the original User to
to avoid what would be a confusing error message.

Don't try AcquireLocalTicket at all against vCenter as it is unsupported.

Fixes #952

### CODE REVIEW

File: govc/flags/client.go [hunk 1]
```diff
@@ -428,11 +428,15 @@ func (flag *ClientFlag) newClient() (*vim25.Client, error) {
...
 	u := flag.url.User
 
-	if u.Username() == "" {
-		// Assume we are running on an ESX or Workstation host if no username is provided
-		u, err = flag.localTicket(ctx, m)
-		if err != nil {
-			return nil, err
+	if u.Username() == "" && !c.IsVC() {
+		// If no username is provided, try to acquire a local ticket.
+		// When invoked remotely, ESX returns an InvalidRequestFault (VC returns NotSupportedFault).
```
Comment (bradfitz) [inline govc/flags/client.go]: this parenthetical is slightly confusing considering the conditional on line 431 excludes this path for VC.

elaborate or remove?

**Comment by bradfitz** [comment]
Code LGTM. Thanks for the fix!

File: govc/flags/client.go (hunk 1 above)
Comment (dougm) [reply govc/flags/client.go]: Removed.  Thanks for reviewing Brad!

File: govc/flags/client.go [hunk 2]
```diff
@@ -428,11 +428,15 @@ func (flag *ClientFlag) newClient() (*vim25.Client, error) {
...
-	if u.Username() == "" {
-		// Assume we are running on an ESX or Workstation host if no username is provided
-		u, err = flag.localTicket(ctx, m)
-		if err != nil {
-			return nil, err
+	if u.Username() == "" && !c.IsVC() {
+		// If no username is provided, try to acquire a local ticket.
+		// When invoked remotely, ESX returns an InvalidRequestFault.
+		// So, rather than return an error here, fallthrough to Login() with the original User to
+		// to avoid what would be a confusing error message.
```
Comment (anfernee) [inline govc/flags/client.go]: so error happens here, we login with empty user, and get something like `LoginFault`, which is less confusing. Is that accurate?

Comment (dougm) [reply govc/flags/client.go]: exactly, InvalidLoginFault with the appropriate localized message.

**Pull request merged**